scripts/gen_ta_src.py

```diff
@@ -0,0 +1,142 @@
+    i = 0
+    for uuid in uuids:
+        uuid_str = "_".join(uuid)
+        file.write(f'\t{tablename}[{i}].ta_start = __ta_{uuid_str}_start;\n')
+        file.write(f'\t{tablename}[{i}].ta_size = __ta_{uuid_str}_size;\n')
+        i += 1
+    file.write(f'\n\tTEEC_SetTATable({tablename});\n\treturn 0;\n')
+    file.write("}\n\n")
+    file.write('SYS_INIT(init_table, APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);')
+    file.close()
+
+def genarate_asm(uuids, output):
+    filename=output+'.S'
+    file = open(filename, 'w')
+    file.write('/* Do not modify. This is generated file*/\n')
+    file.write('#include <zephyr/toolchain.h>\n\n')
+    for uuid in uuids:
+        uuid_str = "_".join(uuid)
+        file.write(f'.section .data.ta_{uuid_str},"a"\n')
+        file.write(f'GDATA(__ta_{uuid_str}_start)\n')
+        file.write(f'GDATA(__ta_{uuid_str}_size)\n')
+        file.write(f'__ta_{uuid_str}_start:\n')
+        file.write(f'.align 8\n')
+        file.write(f'.incbin  "{"-".join(uuid)}.ta"\n')
+        file.write(f'__ta_{uuid_str}_size: .long . - __ta_{uuid_str}_start\n\n')
+    file.close()
+
+def main():
+    parser = argparse.ArgumentParser(description="Generating source files which creates lookup"
+                                     " table of loadable ta's. This table used by tee supplicant"
+                                     " to load ta to OP-TEE.")
+    parser.add_argument("-u", "--uuids", nargs="*", help="List of enabled layers")
+    parser.add_argument("-f", "--file", action="store", help="Config file with enabled TA's")
+    parser.add_argument("-o", "--out", action='store',
+                        help="Output file name (without extension)", default="output")
+    parser.add_argument("-s", "--source", action='store', help="Directory with prebuilt ta")
+    parser.add_argument("-t", "--table", action='store',
+                        help="table variable name", default="table")
+    args = parser.parse_args()
+    if args.uuids is not None or args.file is not None or args.source is not None:
+        uuids = []
+        if args.uuids is not None:
+            uuids = [get_components(u) for u in args.uuids]
+        if args.file is not None:
+            try:
+                with open(args.file, "r", encoding="utf-8") as enabled_file:
+                    uuids.extend(read_uuids_file(enabled_file))
+            except FileNotFoundError:
+                print(f"File {args.file} can't be opened")
+        if args.source:
+            listdir = os.listdir(args.source)
+            for item in listdir:
+                if item.endswith(".ta"):
+                    uuid = os.path.splitext(os.path.split(item)[1])[0]
+                    components = get_components(uuid)
+                    found = False
+                    for i in uuids:
+                        if compare_uuids(i, components):
+                            found = True
+                            break;
+                    if not found:
```

Comment:
You can use `else:` in `for` loop. This will allow you to get rid of `found` variable

Reply:
This was refactored. Got rid from for loop